hbase-client/src/main/java/org/apache/hadoop/hbase/client/RpcRetryingCallerFactory.java

```diff
@@ -71,30 +71,41 @@ public <T> RpcRetryingCaller<T> newCaller() {
       interceptor, startLogErrorsCnt, connectionConf.getRpcTimeout(), metrics);
   }
 
-  public static RpcRetryingCallerFactory instantiate(Configuration configuration,
+  @RestrictedApi(explanation = "Should only be called on process initialization", link = "",
+    allowedOnPath = ".*/hbase-server/src/main/java/.*/(HRegionServer|LoadIncrementalHFiles|SecureBulkLoadClient)\\.java")
+  public static RpcRetryingCallerFactory instantiate(Configuration configuration, MetricsConnection metrics) {
+    return instantiate(configuration, new ConnectionConfiguration(configuration), metrics);
+  }
+
+  public static RpcRetryingCallerFactory instantiate(Configuration configuration, ConnectionConfiguration connectionConf,
     MetricsConnection metrics) {
-    return instantiate(configuration, RetryingCallerInterceptorFactory.NO_OP_INTERCEPTOR, null,
+    return instantiate(configuration, connectionConf, RetryingCallerInterceptorFactory.NO_OP_INTERCEPTOR, null,
       metrics);
   }
 
-  public static RpcRetryingCallerFactory instantiate(Configuration configuration,
+  public static RpcRetryingCallerFactory instantiate(Configuration configuration, ConnectionConfiguration connectionConf,
     ServerStatisticTracker stats, MetricsConnection metrics) {
-    return instantiate(configuration, RetryingCallerInterceptorFactory.NO_OP_INTERCEPTOR, stats,
+    return instantiate(configuration, connectionConf, RetryingCallerInterceptorFactory.NO_OP_INTERCEPTOR, stats,
       metrics);
   }
 
-  public static RpcRetryingCallerFactory instantiate(Configuration configuration,
+  public static RpcRetryingCallerFactory instantiate(Configuration configuration, ConnectionConfiguration connectionConf,
     RetryingCallerInterceptor interceptor, ServerStatisticTracker stats,
     MetricsConnection metrics) {
     String clazzName = RpcRetryingCallerFactory.class.getName();
     String rpcCallerFactoryClazz =
       configuration.get(RpcRetryingCallerFactory.CUSTOM_CALLER_CONF_KEY, clazzName);
     RpcRetryingCallerFactory factory;
     if (rpcCallerFactoryClazz.equals(clazzName)) {
```

Comment:
I suppose we only want to use this in tests? Not for users...

Reply:
Hmm yes, I suppose.